tests/src/ut_test_control_plane.c

```diff
@@ -0,0 +1,169 @@
+    wait_for_1_second();
+
+    UT_ControlPlane_Exit(pInstance);
+    wait_for_1_second();
+
+    UT_ControlPlane_Exit(pInstance1);
+    wait_for_1_second();
+
+}
+
+void test_ut_control_testService()
+{
+    ut_controlPlane_instance_t *pInstance = NULL;
+    ut_controlPlane_instance_t *pInstance1 = NULL;
+
+    pInstance = UT_ControlPlane_Init(8080);
+    UT_ControlPlane_Start(NULL);
+    UT_ControlPlane_Start(pInstance);
+
+    wait_for_1_second();
+    pInstance1 = UT_ControlPlane_Init(9000);
+    UT_ControlPlane_Start(pInstance1);
+
+    wait_for_1_second();
+    UT_ASSERT( pInstance != pInstance1 );
+
+    UT_ControlPlane_Exit(pInstance);
+    UT_ASSERT( pInstance != pInstance1 );
+
+    UT_ASSERT( pInstance != NULL );
+    wait_for_1_second();
+
+    UT_ControlPlane_Exit(pInstance1);
+    UT_ASSERT( pInstance != pInstance1 );
+}
+
+void testCallback(char *key, ut_kvp_instance_t *instance)
+{
+    printf("*******************************Inside testCallback************************\n");
+    ut_kvp_print( instance );
+}
+
+void testRMFCallback(char *key, ut_kvp_instance_t *instance)
+{
+    UT_LOG("\n**************testRMFCallback is called****************\n");
+    ut_kvp_print( instance );
+}
+
+static void UT_ControlPlane_Sigint_Handler(int sig)
+{
+    wait_for_1_second();
+    UT_ControlPlane_Exit(gInstance);
+}
+
+void test_ut_control_performInit( void )
+{
+    gInstance = UT_ControlPlane_Init(8080);
+    UT_ASSERT(gInstance != NULL);
+
+    signal(SIGINT, UT_ControlPlane_Sigint_Handler);
+}
+
+void test_ut_control_performStart()
+{
+
+    UT_ControlPlane_RegisterCallbackOnMessage(gInstance, "test/myCommand", &testCallback);
+    UT_LOG("\n Please run this command from the command line");
+    UT_LOG("\n Waiting client test/myCommand to send yaml/json file");
+
+    UT_ControlPlane_RegisterCallbackOnMessage(gInstance, "test2/myCommand1", &testRMFCallback);
+    UT_LOG("\nWaiting client test/myCommand2 to send yaml/json file");
+
+    UT_ControlPlane_Start(gInstance);
+}
+
+void wait_function()
+{
+    for(int i = 0; i < 20; i++)
+    {
+        UT_LOG("Waiting for %d sec", i);
+        sleep(1);
+    }
+}
+
+void test_ut_control_performExit( void )
+{
+    signal(SIGINT, NULL);
+    UT_ControlPlane_Exit(gInstance);
+}
+
+void register_cp_function()
+{
+    /* Testing my functions work ok */
+    gpAssertSuite1 = UT_add_suite("ut-cp - system control", NULL, NULL);
+    assert(gpAssertSuite1 != NULL);
+    UT_add_test(gpAssertSuite1, "ut-cp Init Exit", test_ut_control_testInitExit);
+    UT_add_test(gpAssertSuite1, "ut-cp websocket service", test_ut_control_testService);
+
+    /* Testing the functionality */
+    gpAssertSuite2 = UT_add_suite("ut-cp - client control", NULL, NULL);
+    assert(gpAssertSuite2 != NULL);
+    UT_add_test(gpAssertSuite2, "ut-cp Init", test_ut_control_performInit);
+    UT_add_test(gpAssertSuite2, "ut-cp Start", test_ut_control_performStart);
+    UT_add_test(gpAssertSuite2, "ut-cp Wait", wait_function);
```

Comment:
that's not a valid test.. you must NEVER wait, you must only expect information to arrive, which can be checked , or timeout.

Reply:
As discussed, received message:

```
curl -F file=@/home/jpn323/workspace/ut-core/framework/ut-control/tests/websocket-clients/example.yaml ws://localhost:8080
curl: (1) Protocol "ws" not supported or disabled in libcurl

```

Reply:
https://github.com/rdkcentral/ut-control/issues/3